Changed casting for write offset.

bgiorgio0506 opened the pull request:
I have changed the casting for type Int64  from Int32Value to Int64Value. This casting was creating an issue while writing on Int64 offset types.

I have tested the code as part of the process of validating changes for another porject. 
Also to note the Typescript definitions are wrong for types Int64 and Uint64 as the function requires 4 arguments for those types and not 3 ( I will open a new PR for TS types).

koesie10 reviewed (commented):
Thanks! Could you also make the equivalent change on line 343 for uint64?

bgiorgio0506 commented:
> Thanks! Could you also make the equivalent change on line 343 for uint64?

yes sure I will do it in the coming weeks ; )

bgiorgio0506 commented:
@koesie10 you can merge if my changes satisfy your code standards

koesie10 commented on packages/fsuipc/src/FSUIPC.cc:
```diff
@@ -340,7 +340,7 @@ void FSUIPC::Write(const Napi::CallbackInfo& info) {
         x = std::stoll(x_str);
       } else if (info[3].IsNumber()) {
-        x = (uint64_t)info[3].ToNumber().Uint32Value();
+        x = info[3].As<Napi::BigInt>().Uint64Value(false);
```
Why do we need to create a bigint before we're able to get the uint64 value? I would expect this to work the same as the `Int64Value`.

bgiorgio0506 replied:
@koesie10  this is bc uint64 is not a member of Number you can cast it only through BigInt  see Napi doc https://github.com/nodejs/node-addon-api/blob/main/doc/bigint.md and https://github.com/nodejs/node-addon-api/blob/main/doc/number.md

koesie10 commented on package-lock.json:
Can you please remove this file and revert the changes to `yarn.lock`?

bgiorgio0506 replied:
yes sure will do

koesie10 reviewed (commented):
Thanks again for your contribution and sorry for my late response. I still have a question and a suggestion before this can be merged.

bgiorgio0506 commented:
@koesie10 sorry for the delay I have reverted the yarn.lock file and removed the package-lock.json. I have answerd your previous question let me know if you need some more clarification

koesie10 commented:
@bgiorgio0506 Thanks! I've cherry-picked your changes in #33 and made some additional changes such that the types are now correct.

koesie10 commented:
@bgiorgio0506 This has been published in version v0.7.2.

bgiorgio0506 commented:
@koesie10  thanks !
